File: app/src/main/java/yangfentuozi/runner/service/callback/ExecResultCallback.java
package yangfentuozi.runner.service.callback;

import android.os.RemoteException;

public class ExecResultCallback {

    private final IExecResultCallback mCallback;

    public ExecResultCallback(IExecResultCallback callback) {
        mCallback = callback;
    }

    public void onOutput(String outputs) {
        try {
            mCallback.onOutput(outputs);
        } catch (RemoteException ignored) {
        }
    }

    public void onExit(int exitValue) {
        try {
            mCallback.onExit(exitValue);
        } catch (RemoteException ignored) {
        }
    }
}

File: app/src/main/java/yangfentuozi/runner/server/callback/ExecResultCallback.kt
package yangfentuozi.runner.server.callback

import android.os.RemoteException

class ExecResultCallback(private val mCallback: IExecResultCallback?) {
    fun onOutput(outputs: String?) {
        try {
            mCallback?.onOutput(outputs)
        } catch (_: RemoteException) {
        }
    }

    fun onExit(exitValue: Int) {
        try {
            mCallback?.onExit(exitValue)
        } catch (_: RemoteException) {
        }
    }
}
